Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+        void CreateInstance(object pUnkOuter, ref Guid riid, out object ppvObj);
+        void GetMops(int memid, out string pBstrMops);
+        void GetContainingTypeLib(out IntPtr ppTLB, out int pIndex);
+        void ReleaseTypeAttr(IntPtr pTypeAttr);
+        void ReleaseFuncDesc(IntPtr pFuncDesc);
+        void ReleaseVarDesc(IntPtr pVarDesc);
+
+        void Placeholder1();
+        IDispatch GetStdModInstance();            // a handy extra vtable entry we can use to invoke members in standard modules.
+    }
+
+    // FIXME there's probably some better builtin c# class for this
+    public class DisposableList<T> : List<T>, IDisposable
+        where T : class
+    {
+        public void Dispose() => ((IDisposable)this).Dispose();
+
+        private bool _isDisposed;
+        void IDisposable.Dispose()
+        {
+            if (_isDisposed) return;
+            _isDisposed = true;
+
+            foreach (var element in this)
+            {
+                ((IDisposable)element).Dispose();
+            }
+        }
+    }
+
+    public enum TypeLibConsts : int
+    {
+        MEMBERID_NIL = -1,
+    }
+
+    // A wrapper for ITypeInfo provided by VBE, allowing safe managed consumption, plus adds StdModExecute functionality
+    public class TypeInfoWrapper_VBE : ITypeInfo, IDisposable
+    {
+        private DisposableList<TypeInfoWrapper_VBE> _typeInfosWrapped;
+        private DisposableList<TypeLibWrapper_VBE> _typeLibsWrapped;
+        private AggregateSingleInterface<ITypeInfo> _typeInfoAggregatorObj;
+        private bool _isUserFormBaseClass = false;
+        private int _firstImplementedTypeHref = -1;
+        private TYPEATTR _cachedAttributes;
+
+        public readonly string Name;
+        public readonly string DocString;
+        public readonly int HelpContext;
+        public readonly string HelpFile;
+
+        private ITypeInfo _wrappedObject;
+        private ITypeInfo_VBE _ITypeInfoAlt
+        { get => ((ITypeInfo_VBE)_wrappedObject); }
+
+        public TypeInfoWrapper_VBE(IntPtr rawObjectPtr, bool isBrokenUserFormBaseClass = false, bool isBrokenUserFormBaseEventsClass = false)
+        {
+            _typeInfoAggregatorObj = new AggregateSingleInterface<ITypeInfo>(rawObjectPtr);
+            var aggObjPtr = Marshal.CreateAggregatedObject(rawObjectPtr, _typeInfoAggregatorObj);
+            _wrappedObject = (ITypeInfo)Marshal.GetObjectForIUnknown(aggObjPtr);        // when this CCW object gets released, it will free the aggObjInner (well, after GC)
+            Marshal.Release(aggObjPtr);         // _wrappedObject holds a reference to this now
+
+            IntPtr typeAttrPtr = IntPtr.Zero;
+            GetTypeAttr(out typeAttrPtr);
+            _cachedAttributes = StructHelper.ReadStructure<TYPEATTR>(typeAttrPtr);
+            ReleaseTypeAttr(typeAttrPtr);      // don't need to keep a hold of it, as _cachedAttributes is a copy
+
+            if (isBrokenUserFormBaseClass)
```

Comment:
I think this needs more comment as I'm not sure how exactly it's "broken". And is it really only the UF? I feel that those kind of checks should be handled somewhere else - maybe a dedicated helper static class so that we can add additional functions for different types?

Reply:
See comment about UserForm being broken in top class comment.  As Vogel says, some of the comments need distributing better

Reply:
The UserForm base class is generated dynamically by VBE at runtime, and when you try to obtain its name using GetDocumentation(), it crashes with an access violation, presumably because they chose to not bother naming something that was never designed to be known about